redash/handlers/dashboards.py

```diff
@@ -8,6 +11,19 @@
 from redash.handlers.base import BaseResource, get_object_or_404
 
 
+def _save_change(user, dashboard_id, old_dashboard, new_dashboard, change_type):
```

Comment:
I think we can generalize this and have a single method for all change-tracking objects. Also it should be at the model level.

Ideally, it will be a hook that is being called on the model's `#save` method, but at that point we don't know who the user who made the change :-\ Maybe we can add another save method named `tracked_save` that will also take as a parameter the user who made the change.

One benefit of doing this at the model level, is that we can leverage `dirty_fields` property of the model (that will return the fields that were changed).